### PR TITLE
[MIN-315] Implement get_utilization_rate RPC

### PR DESCRIPTION
**Pull request by dev-vadim**
**Description:**

Added RPC that returns current utilization rate of the pool

**Changes in storage or API:**

Controller RPC:
 - add RPC getUtilizationRate

**Checklist**

- [x] Code is covered with tests
- [x] Code is covered with benchmarks
- [x] Fresh benchmark results included

### CODE REVIEW

File: pallets/controller/src/lib.rs [hunk 1]
```diff
@@ -797,6 +802,25 @@ impl<T: Config> Pallet<T> {
...
+		let current_block_number = <frame_system::Module<T>>::block_number();
+		let accrual_block_number_previous = Self::controller_dates(pool_id).last_interest_accrued_block;
+		let block_delta = Self::calculate_block_delta(current_block_number, accrual_block_number_previous)?;
+		let pool_data = Self::calculate_interest_params(pool_id, block_delta)?;
+		let current_total_balance = T::LiquidityPoolsManager::get_pool_available_liquidity(pool_id);
+		Self::calculate_utilization_rate(
+			current_total_balance,
+			pool_data.total_borrowed,
+			pool_data.total_protocol_interest,
+		)
```
Comment (lut1k) [inline pallets/controller/src/lib.rs:821]: I think we can implement this functionality in the method `pub fn get_utilization_rate(pool_id: CurrencyId)`. And this method is redundant, in my opinion.

File: runtime/src/tests/rpc_tests.rs [hunk 2]
```diff
@@ -273,6 +273,43 @@ fn test_rates_using_rpc() {
...
+			// No borrows -> utilization rates equal to 0
+			assert_eq!(get_utilization_rate_rpc(DOT), Some(Rate::zero()));
+			assert_eq!(get_utilization_rate_rpc(ETH), Some(Rate::zero()));
+
+			System::set_block_number(20);
+
+			assert_ok!(MinterestProtocol::borrow(bob(), DOT, dollars(100_000)));
+			assert_ok!(MinterestProtocol::repay(bob(), DOT, dollars(30_000)));
+			assert_eq!(pool_balance(DOT), dollars(80_000));
+			// 70 / 80 + 70 = 0.466666667
```
Comment (lut1k) [inline runtime/src/tests/rpc_tests.rs:304]:
```suggestion
			// 70 / (80 + 70) = 0.466666667
```

File: pallets/controller/src/lib.rs (hunk 1 above)
Comment (dev-vadim) [reply pallets/controller/src/lib.rs:821]: In my opinion it`s nice that RPC method implementation just calls inner function

File: runtime/src/tests/rpc_tests.rs (hunk 2 above)
Comment (dev-vadim) [reply runtime/src/tests/rpc_tests.rs]: Done

File: runtime/src/tests/rpc_tests.rs [hunk 3]
```diff
@@ -273,6 +273,43 @@ fn test_rates_using_rpc() {
 		});
 }
 
+#[test]
+fn test_get_utilization_rate_rpc() {
```
[the remark this answers is not in the document]
Comment (dev-vadim) [reply runtime/src/tests/rpc_tests.rs:277]: Done

File: pallets/controller/src/lib.rs (hunk 1 above)
Comment (Apollo50) [reply pallets/controller/src/lib.rs:821]: In my view, following our implicit convention, we implement inner Rpc logic directly in rpc block. If it used for other fns, we could move it.
In this case i agree with @lut1k

File: runtime/src/tests/rpc_tests.rs [hunk 4]
```diff
@@ -273,6 +273,51 @@ fn test_rates_using_rpc() {
...
+			assert_ok!(MinterestProtocol::enable_is_collateral(bob(), DOT));
+			assert_ok!(MinterestProtocol::enable_is_collateral(bob(), ETH));
+			// No borrows -> utilization rates equal to 0
+			assert_eq!(get_utilization_rate_rpc(DOT), Some(Rate::zero()));
+			assert_eq!(get_utilization_rate_rpc(ETH), Some(Rate::zero()));
+
+			System::set_block_number(20);
+
+			assert_ok!(MinterestProtocol::borrow(bob(), DOT, dollars(100_000)));
+			assert_ok!(MinterestProtocol::repay(bob(), DOT, dollars(30_000)));
```
Comment (kurotych) [inline runtime/src/tests/rpc_tests.rs:302]: Why do we actually need this actions(repay) in this test?

Comment (dev-vadim) [reply runtime/src/tests/rpc_tests.rs:302]: No, we can achieve the same by borrowing 70_000

Comment (dev-vadim) [reply runtime/src/tests/rpc_tests.rs]: Done

File: pallets/controller/src/lib.rs (hunk 1 above)
Comment (dev-vadim) [reply pallets/controller/src/lib.rs]: Done

**Review by lut1k — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Review by lut1k — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**